supervisor/shared/tick.c

```diff
@@ -105,6 +105,12 @@ void mp_hal_delay_ms(mp_uint_t delay) {
         if(MP_STATE_VM(mp_pending_exception) == MP_OBJ_FROM_PTR(&MP_STATE_VM(mp_kbd_exception)) ||
            MP_STATE_VM(mp_pending_exception) == MP_OBJ_FROM_PTR(&MP_STATE_VM(mp_reload_exception)) ||
            WATCHDOG_EXCEPTION_CHECK()) {
+           if(MP_STATE_VM(mp_pending_exception) == MP_OBJ_FROM_PTR(&MP_STATE_VM(mp_kbd_exception)))
+           {
+            // clear exception and generate stacktrace
+            MP_STATE_VM(mp_pending_exception) = MP_OBJ_NULL;
+            nlr_raise(mp_obj_new_exception(&mp_type_KeyboardInterrupt));
+           }
```

Comment:
Could you simplify this? It ends up checking `MP_STATE_VM(mp_pending_exception) == MP_OBJ_FROM_PTR(&MP_STATE_VM(mp_kbd_exception))` twice. In that case we want to clear the exception and raise `KeyboardInterrupt` (which will leave the loop immediately). If it's not an `mp_kbd_exception`, we just want to break. I think:
```c
        if (MP_STATE_VM(mp_pending_exception) == MP_OBJ_FROM_PTR(&MP_STATE_VM(mp_kbd_exception)))         {
￼            // clear exception and generate stacktrace
￼            MP_STATE_VM(mp_pending_exception) = MP_OBJ_NULL;
￼            nlr_raise(mp_obj_new_exception(&mp_type_KeyboardInterrupt));
￼           }
        if (MP_STATE_VM(mp_pending_exception) == MP_OBJ_FROM_PTR(&MP_STATE_VM(mp_reload_exception)) ||
￼           WATCHDOG_EXCEPTION_CHECK()) {
            // stop sleeping immediately
            break;
        }
```